contract/contracts/Voting.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+import "./Types.sol";
+
+contract Voting is OwnableUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+
+    mapping(uint256 => Types.Proposal) public proposals;
+    uint256 public proposalCount;
+
+    event ProposalAdded(uint256 indexed proposalId, string title, string description);
+    event DirectiveVoted(address indexed voter, uint256 indexed proposalId, uint256 vote);
+    event ElectionVoted(address indexed voter, uint256 indexed proposalId, address indexed candidateAddress);
+    event ProposalConcluded(uint256 indexed proposalId, Types.Proposal proposal);
+
+    function initialize() public initializer {
+        __Ownable_init(msg.sender);
+    }
+
+   function addProposal(Types.Proposal calldata _proposal) public onlyOwner whenNotPaused {
+        proposals[proposalCount] = _proposal;
+        emit ProposalAdded(proposalCount, _proposal.title,_proposal.description);
+        proposalCount++;
+    }
```

Comment:
@hermannleboss is it possible to pass Struct into an function argument?

Reply:
Yes, you can. You can check how I'm calling it from the Voting.spec.ts
``` 
const proposal = {
          title: 'Proposal 1',
          description: 'Description of Proposal 1',
          draftedBy: await owner.getAddress(),
          isElection: false,
          votes: { yes: 0, no: 0, abstain: 0 },
          candidates,
          isActive: true,
          voters: [
            {
              name: 'Member 1',
              memberAddress: await member1.getAddress(),
              isVoted: false,
              isEligible: true
            },
            {
              name: 'Member 2',
              memberAddress: await member2.getAddress(),
              isVoted: false,
              isEligible: true
            }
          ]
        }

        await expect(votingProxy.addProposal(proposal))
          .to.emit(votingProxy, 'ProposalAdded')
          .withArgs(0, proposal.title, proposal.description)

Reply:
great catch! in that case, we can manually set each of the properties and assign 0 initially during the creation of the proposal in the Solidity function:

```
proposals[proposalCount].title = _proposal.title
//..
//...
// and so on
proposal[proposalCount].votes={yes:0,no:0,abstain:0};
 ```